judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/AbstractTest.java

```diff
@@ -0,0 +1,538 @@
+        hDao.queryRelationIOnHMulti(h).execute().stream().forEach(i -> assertEquals(Optional.of("iMultiRelationNameChanged"), i.getNameI()));
+
+        //delete with IDao
+
+        iDao.delete(h.getCompositionIOnHSingle().get());
+        assertTrue(iDao.getById(h.getCompositionIOnHSingle().get().identifier()).isEmpty());
+        assertTrue(jDao.getById(h.getCompositionIOnHSingle().get().identifier().adaptTo(JIdentifier.class)).isEmpty());
+        assertEquals(iDao.getAll().size(),jDao.getAll().size());
+
+        //TODO JNG-4812
+//        iDao.delete(h.getCompositionIOnHSingleRequired()); //unfortunately you can delete required Component
+//        h = hDao.getById(h.identifier()).orElseThrow();
+
+        iDao.delete(h.getCompositionIOnHMulti().get(0));
+        assertTrue(iDao.getById(h.getCompositionIOnHMulti().get(0).identifier()).isEmpty());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(2,h.getCompositionIOnHMulti().size());
+
+        //check the other methods are work well
+
+        //add
+        hDao.addCompositionIOnHMulti(h, Collections.singletonList(JtoI(jDao.create(J.builder().build()))));
+        assertEquals(2, h.getCompositionIOnHMulti().size());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(3, h.getCompositionIOnHMulti().size());
+
+        assertEquals(3, hDao.countRelationIOnHMulti(h));
+        hDao.addRelationIOnHMulti(h, List.of(JtoI(jDao.create(J.builder().build()))));
+        assertEquals(4, hDao.countRelationIOnHMulti(h));
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        //remove
+        hDao.removeCompositionIOnHMulti(h, List.of(h.getCompositionIOnHMulti().get(0)));
+        assertEquals(3,h.getCompositionIOnHMulti().size());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(2,h.getCompositionIOnHMulti().size());
+
+        assertEquals(4,hDao.countRelationIOnHMulti(h));
+        hDao.removeRelationIOnHMulti(h, List.of(hDao.queryRelationIOnHMulti(h).execute().get(0)) );
+        assertEquals(3,hDao.countRelationIOnHMulti(h));
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        hDao.removeCompositionIOnHMulti(h, List.of(h.getCompositionIOnHMulti().get(0),h.getCompositionIOnHMulti().get(1)));
+        assertEquals(2,h.getCompositionIOnHMulti().size());
+        h = hDao.getById(h.identifier()).orElseThrow();
+        assertEquals(0,h.getCompositionIOnHMulti().size());
+
+        assertEquals(3,hDao.countRelationIOnHMulti(h));
+        hDao.removeRelationIOnHMulti(h, List.of(hDao.queryRelationIOnHMulti(h).execute().get(0),hDao.queryRelationIOnHMulti(h).execute().get(1)) );
+        assertEquals(1,hDao.countRelationIOnHMulti(h));
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        //set
+        hDao.setCompositionIOnHSingle(h,JtoI(jDao.create(J.builder().withNameI("SetNewElement").build())));
```

Comment:
> setCompositionIOnHSingle

I think set on a composition should not work.

Isn't there a TBD issue about this? If not I suggest creating one:D

Reply:
[6f85e57](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/165/commits/6f85e57884996176f05ef4554ae4669dee9e50a2)